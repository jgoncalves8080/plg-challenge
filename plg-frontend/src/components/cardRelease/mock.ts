import { CardReleaseProps } from './types'

export const mockLauch: CardReleaseProps = {
  name: 'Starlink 4-35 (v1.5)',
  details: 'Lorem ipsum',
  links: {
    patch: {
      small: 'https://images2.imgbox.com/a9/9a/NXVkTZCE_o.png',
      large: 'https://images2.imgbox.com/e3/cc/hN96PmST_o.png'
    }
  }
}
