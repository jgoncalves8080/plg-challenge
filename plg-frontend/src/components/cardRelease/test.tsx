/**
 * @jest-environment jsdom
 */
import { render } from '../../utils/tests/wrapper'
import { CardRelease } from '.'
import { mockLauch } from './mock'

describe('CardRelease Component', () => {
  it('renders correctly', () => {
    const { container, getByText } = render(<CardRelease {...mockLauch} />)
    expect(container.firstChild).toMatchSnapshot()

    expect(getByText(mockLauch.name))
  })

  it('should render the information received via props', () => {
    const { getAllByRole, getByText } = render(<CardRelease {...mockLauch} />)

    expect(getAllByRole('img')[0]).toHaveAttribute(
      'src',
      mockLauch.links.patch.large
    )

    expect(getAllByRole('img')[1]).toHaveAttribute(
      'src',
      mockLauch.links.patch.small
    )

    expect(getByText(mockLauch.name)).toBeInTheDocument()
    expect(getByText(mockLauch.details)).toBeInTheDocument()
  })
})
