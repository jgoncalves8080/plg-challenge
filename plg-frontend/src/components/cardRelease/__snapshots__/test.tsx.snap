// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardRelease Component renders correctly 1`] = `
<header
  class="styles__Container-sc-1u0rdqk-0"
>
  <div
    class="styles__Card-sc-1u0rdqk-3 brPiqr"
  >
    <img
      alt="Starlink 4-35 (v1.5)"
      class="styles__CardImage-sc-1u0rdqk-4 jpbJCI"
      src="https://images2.imgbox.com/e3/cc/hN96PmST_o.png"
    />
    <div
      class="styles__CardOverlay-sc-1u0rdqk-1 gfeRBZ"
    >
      <div
        class="styles__CardHeader-sc-1u0rdqk-2 esiKUe"
      >
        <svg
          class="styles__CardArc-sc-1u0rdqk-5 iVbTlg"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path />
        </svg>
        <img
          alt="Starlink 4-35 (v1.5)"
          class="styles__CardThumb-sc-1u0rdqk-6 cXYgJW"
          src="https://images2.imgbox.com/a9/9a/NXVkTZCE_o.png"
        />
        <div
          class="styles__CardHeaderText-sc-1u0rdqk-7 ehbYzA"
        >
          <h3
            class="styles__CardTitle-sc-1u0rdqk-8 fZgplc"
          >
            Starlink 4-35 (v1.5)
          </h3>
        </div>
      </div>
      <p
        class="styles__CardDescription-sc-1u0rdqk-9 fKkkHG"
      >
        Lorem ipsum
      </p>
    </div>
  </div>
</header>
`;
